### PR TITLE
Reload functionality

### PR DESCRIPTION
**Pull request by btzr-io**
### Changes
- Reload current page and clear the cache.
- Add reload button inside header component.
- Fixed issue with `Published` page after clearing the cache: https://github.com/lbryio/lbry-app/pull/682#issuecomment-337434699

See: #672

### CODE REVIEW

**Comment by kauffj** [comment]
I kind of dislike adding this, because on one hand, this should never be necessary.

OTOH, it probably is currently, and there's this argument: https://www.tbray.org/ongoing/When/201x/2017/09/27/Refresh-Button

If we're going to give this such prominent placement, should it perhaps clear the cache as well? If a user wants to reload, it's likely because they are concerned about state being stale, and the application cache is a potential source of that.

**Comment by kauffj** [comment]
@btzr-io any reason you closed this? My comment wasn't a no :)

**Comment by btzr-io** [comment]
>  This should never be necessary. 

if shouldn't be necessary then why we add it? :confused:  ^^

>  should it perhaps clear the cache as well? 

Would this clear the history too? , it should reload the current page..

**Comment by kauffj** [comment]
> if shouldn't be necessary then why we add it? :confused: ^^

_Ideally_ it is not necessary. Pragmatically it may be. Check out the link, it's a solid argument.

I'm not 100% sure on cache clear behavior, but I agree that if it's clearing the history it's a bad idea.

**Comment by btzr-io** [comment]
Also after clearing the cache, the publishes page shows:
> It looks like you haven't published anything to LBRY yet.

This could  cause some confusion since there is no indication of fetching / loading any data and looks like all your claims are gone :stuck_out_tongue:

**Comment by kauffj** [comment]
If you also invoke the actions in this function I bet state is properly restored: https://github.com/lbryio/lbry-app/blob/b8980c00d363ea2fffc160ed9471c20bb4079127/ui/js/actions/app.js#L177

**Comment by kauffj** [comment]
(which ought to be refactored if invoked by refresh)

**Comment by btzr-io** [comment]
### Note UX
Would be nice to have an indication if the page  it's reloadign, right now looks like if the app crashed.( empty green screen )

**Comment by btzr-io** [comment]
I guess we only need -> ` dispatch(doFetchFileInfosAndPublishedClaims());`

**Comment by btzr-io** [comment]
@kauffj yeah it works :smiley:  ^

**Comment by btzr-io** [comment]
Do we need this?
https://github.com/lbryio/lbry-app/blob/782fb5d59ebbe6934f8ec97b9f2e77653cec2cb4/ui/js/page/settings/view.jsx#L26

**Comment by btzr-io** [comment]
I'll check this later...
https://github.com/electron/electron/blob/master/docs/api/session.md#sesclearcachecallback
https://github.com/electron/electron/blob/master/docs/api/session.md#sesclearstoragedataoptions-callback

**Comment by btzr-io** [comment]
> I'm not 100% sure on cache clear behavior
I added it in the reload function, didn't find any issues...
The default behavior or the refresh button it's just "reload current page" see any browser for reference...

File: ui/js/actions/app.js [hunk 1]
```diff
@@ -200,11 +200,22 @@ export function doRemoveSnackBarSnack() {
...
-
+    dispatch(doFetchFileInfosAndPublishedClaims());
     return Promise.resolve();
   };
 }
 
+export function doReloadCurrentPage() {
+  return function(dispatch, getState) {
+    const currentPage = remote.webContents.getAllWebContents()[0];
+    if (currentPage) {
```
Comment (btzr-io) [inline ui/js/actions/app.js]: not sure if needs this...

File: ui/js/actions/app.js [hunk 2]
```diff
@@ -200,11 +200,22 @@ export function doRemoveSnackBarSnack() {
 export function doClearCache() {
   return function(dispatch, getState) {
     window.cacheStore.purge();
-
+    dispatch(doFetchFileInfosAndPublishedClaims());
```
[the remark this answers is not in the document]
Comment (btzr-io) [reply ui/js/actions/app.js]: weird, the issue is back ^

**Comment by btzr-io** [comment]
This should be fixed before adding this...
see: https://github.com/lbryio/lbry-app/pull/682#issuecomment-337452542 and https://github.com/lbryio/lbry-app/pull/682#issuecomment-337434699

**Comment by btzr-io** [comment]
Also https://github.com/lbryio/lbry-app/pull/688 make this unnecessary.

**Comment by kauffj** [comment]
Agreed, this is no longer needed with #688

**Pull request closed without merging**